### PR TITLE
[Docs] Add Materialized view manual

### LINKED ISSUE #3344: [Proposal] Materialized View 2.0
# The status of materialized view 1.0

In the present, we have supported the materialized views in Doris 0.12 version. The materialized view selector supports to select the most efficient mv and rewrite the SQL to query against the selected mv instead of the base table.
For query results contain a small number of rows where the original table has a large amount of data, the performance can reach the 5X to 100X times depends on the cardinality of the data.
The aggregate functions supported by the materialized view in 0.12 include: sum, min, max.

However, the aggregate functions supported by the current materialized view are not rich enough to fully cover the user's scene.
For example, in the `Order` scenario, user needs to analyze the number of orders in different dimensions.
Another example is the `count_distinct` function is used for analyzing PV and UV data in website traffic.

# The goal of materialized view 2.0

In order to support more scenarios, the materialized view 2.0 will support the following functions:

1. Materialized view supports aggregate functions: count, count_distinct (bitmap and hll)
2. Support to create materialized views of the same column with different aggregate functions. For example: ```select k1, sum (v1), min (v1) from table group by k1```

### PR DESCRIPTION
**Pull request by EmmyMiao87**
Change-Id: Id7dc1a1e0f7b8e13ac12c9524b7b8f3ca5814873

## Proposed changes

Add usage manual of materialized view in Chinese and English

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [] Bugfix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation Update (if none of the other choices apply)
- [] Code refactor (Modify the code structure, format the code, etc...)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [] I have create an issue on #ISSUE, and have described the bug/feature there in detail
- [] Compiling and unit tests pass locally with my changes
- [] I have added tests that prove my fix is effective or that my feature works
- [] If this change need a document change, I have updated the document
- [] Any dependent changes have been merged

### CODE REVIEW

File: docs/zh-CN/administrator-guide/materialized_view.md [hunk 1]
```diff
@@ -0,0 +1,490 @@
...
+```
+HELP CREATE MATERIALIZED VIEW
+```
+
+### 支持聚合函数
+
+目前物化视图功能支持的聚合函数有：
+
++ SUM, MIN, MAX (Version 0.12)
++ COUNT, BITMAP\_UNION, HLL\_UNION (Version 0.13)
```
Comment (wutiangan) [inline docs/zh-CN/administrator-guide/materialized_view.md:92]: 你是说物化视图建表语句支持的聚合函数，还是sql查询中可以匹配的聚合函数？如果是后者的话，物化视图可以支持count(distinct).

File: docs/zh-CN/administrator-guide/materialized_view.md [hunk 2]
```diff
@@ -0,0 +1,490 @@
...
+
+### 更新策略
+
+为保证物化视图表和 Base 表的数据一致性, Doris 会将导入，删除等对 base 表的操作都同步到物化视图表中。并且通过增量更新的方式来提升更新效率。通过事务方式来保证原子性。
+
+比如如果用户通过 INSERT 命令插入数据到 base 表中，则这条数据会同步插入到物化视图中。当 base 表和物化视图表均写入成功后，INSERT 命令才会成功返回。
```
Comment (wutiangan) [inline docs/zh-CN/administrator-guide/materialized_view.md:98]: Remove trailing invisible characters(98 line, 114 line, 328 line, 367 line, 485line )

File: docs/zh-CN/administrator-guide/materialized_view.md [hunk 3]
```diff
@@ -0,0 +1,490 @@
...
+如果用户不再需要物化视图，则可以通过下面命令删除物化视图：
+
+```
+DROP MATERIALIZED VIEW
+```
+
+具体的语法可以通过下面命令查看：
+
+```
+HELP DROP MATERIALIZED VIEW
```
Comment (wutiangan) [inline docs/zh-CN/administrator-guide/materialized_view.md:160]: drop MATERIALIZED VIEW章节有一些不可显字符，建议这个pr也一起修复

File: docs/zh-CN/administrator-guide/materialized_view.md [hunk 4]
```diff
@@ -0,0 +1,490 @@
...
+	|                        |               |                      |             |      |       |         |              |
+	| advertiser_uv          | AGG_KEYS      | advertiser           | VARCHAR(10) | Yes  | true  | NULL    |              |
+	|                        |               | channel              | VARCHAR(10) | Yes  | true  | NULL    |              |
+	|                        |               | to_bitmap(`user_id`) | BITMAP      | No   | false |         | BITMAP_UNION |
+	+------------------------+---------------+----------------------+-------------+------+-------+---------+--------------+
+	```
+	
+2. 查询自动匹配
+
+	当物化视图表创建完成后，查询广告 UV 时，Doris就会自动从刚才创建好的物化视图 `advertiser_uv ` 中查询数据。比如原始的查询语句如下：
```
Comment (wutiangan) [inline docs/zh-CN/administrator-guide/materialized_view.md:364]:
```suggestion
	当物化视图表创建完成后，查询广告 UV 时，Doris就会自动从刚才创建好的物化视图 `advertiser_uv` 中查询数据。比如原始的查询语句如下：
```

File: docs/zh-CN/administrator-guide/materialized_view.md [hunk 5]
```diff
@@ -0,0 +1,490 @@
...
+	
+	这时候查询就会直接从刚才创建的 mv_1 物化视图中读取数据。物化视图对 k3 是存在前缀索引的，查询效率也会提升。
+
+
+## 局限性
+
+1. 物化视图的聚合函数的参数不支持表达式仅支持单列，比如： sum(a+b)不支持。
+2. 如果删除语句的条件列，在物化视图中不存在，则不能进行删除操作。如果一定要删除数据，则需要先将物化视图删除，然后方可删除数据。
+3. 单表上过多的物化视图会影响导入的效率：导入数据时，物化视图和 base 表数据是同步更新的，如果一张表的物化视图表超过10张，则有可能导致导入速度很慢。这就像单次导入需要同时导入10张表数据是一样的。
+
```
Comment (wutiangan) [inline docs/zh-CN/administrator-guide/materialized_view.md:486]: 4. the same column with two aggregate funciton(select sum(a), count(a)) can't be support now
5. aggregation model's table can't support hll_union/bitmap_union/count

File: docs/zh-CN/administrator-guide/materialized_view.md [hunk 6]
```diff
@@ -0,0 +1,490 @@
...
+物化视图中的聚合和查询中聚合的匹配关系：
+
+| 物化视图聚合 | 查询中聚合 |
+| ---------- | -------- |
+| sum        | sum      |
+| min        | min      |
+| max        | max      |
+| count      | count    |
+| bitmap\_union | bitmap\_union, bitmap\_union\_count, count(distinct) |
+| hll\_union | hll\_raw\_agg, hll\_union\_agg, ndv() |
```
Comment (morningman) [inline docs/zh-CN/administrator-guide/materialized_view.md:115]: ndv/approx_count_distinct

File: docs/zh-CN/administrator-guide/materialized_view.md [hunk 7]
```diff
@@ -0,0 +1,490 @@
...
+```
+Query OK, 0 rows affected (0.012 sec)
+```
+
+**第二步：检查物化视图是否构建完成**
+
+由于创建物化视图是一个异步的操作，用户在提交完创建物化视图任务后，需要异步的通过命令检查物化视图是否构建完成。命令如下：
+
+```
+SHOW ALTER TABLE ROLLUP FROM db_name;
```
Comment (morningman) [inline docs/zh-CN/administrator-guide/materialized_view.md:210]: SHOW ALTER TABLE MATERIALIZED VIEW?

File: docs/zh-CN/administrator-guide/materialized_view.md [hunk 8]
```diff
@@ -0,0 +1,490 @@
...
+|           |               | to_bitmap(`k2`) | BITMAP   | No   | false |         | BITMAP_UNION |
+|           |               |                 |          |      |       |         |              |
+| mv_1      | AGG_KEYS      | k4              | SMALLINT | Yes  | true  | NULL    |              |
+|           |               | k1              | BIGINT   | Yes  | false | NULL    | SUM          |
+|           |               | k3              | LARGEINT | Yes  | false | NULL    | SUM          |
+|           |               | k2              | BIGINT   | Yes  | false | NULL    | MIN          |
++-----------+---------------+-----------------+----------+------+-------+---------+--------------+
+```
+
+可以看到当前 `mv_test` 表一共有三张物化视图：mv_1, mv_2 和 mv_3，以及他们的表结构。
```
Comment (morningman) [inline docs/zh-CN/administrator-guide/materialized_view.md:147]: should be `mv\_1` ?

File: docs/zh-CN/administrator-guide/materialized_view.md [hunk 9]
```diff
@@ -0,0 +1,490 @@
...
+	|                        |               | to_bitmap(`user_id`) | BITMAP      | No   | false |         | BITMAP_UNION |
+	+------------------------+---------------+----------------------+-------------+------+-------+---------+--------------+
+	```
+	
+2. 查询自动匹配
+
+	当物化视图表创建完成后，查询广告 UV 时，Doris就会自动从刚才创建好的物化视图 `advertiser_uv ` 中查询数据。比如原始的查询语句如下：
+	
+	```
+	SELECT advertiser, channel, count(distinct user_id) FROM  advertiser_view_record GROUP BY advertiser, channel;
```
Comment (morningman) [inline docs/zh-CN/administrator-guide/materialized_view.md:367]:
```suggestion
	SELECT advertiser, channel, count(distinct user_id) FROM advertiser_view_record GROUP BY advertiser, channel;
```

File: docs/zh-CN/administrator-guide/materialized_view.md (hunk 1 above)
Comment (EmmyMiao87) [reply docs/zh-CN/administrator-guide/materialized_view.md:92]: 物化视图建表支持的聚合函数。

File: docs/zh-CN/administrator-guide/materialized_view.md (hunk 5 above)
Comment (EmmyMiao87) [reply docs/zh-CN/administrator-guide/materialized_view.md:486]: aggregation model's table also support hll_union/bitmap_union/count

**Review by wutiangan — DISMISSED** [review]

File: docs/zh-CN/administrator-guide/materialized_view.md [hunk 10]
```diff
@@ -111,8 +111,8 @@ HELP CREATE MATERIALIZED VIEW
 | min        | min      |
 | max        | max      |
 | count      | count    |
-| bitmap\_union | bitmap\_union, bitmap\_union\_count, count(distinct) |
-| hll\_union | hll\_raw\_agg, hll\_union\_agg, ndv() | 
+| bitmap\_union | bitmap\_union, bitmap\_union\_count, count(distinct) |
+| hll\_union | hll\_raw\_agg, hll\_union\_agg, ndv, approx_count_distinct |
```
Comment (morningman) [inline docs/zh-CN/administrator-guide/materialized_view.md:115]:
```suggestion
| hll\_union | hll\_raw\_agg, hll\_union\_agg, ndv, approx\_count\_distinct | 
```

**Review by morningman — DISMISSED** [review]
LGTM

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**